cluster/gce/windows/common.psm1

```diff
@@ -146,6 +146,307 @@ function MustDownload-File {
+  $file = Get-Item $Path
+  if ($file.Length -gt $Config.MaxSize) {
+    rotate
+    return
+  }
+
+  # Check MaxBackupInterval.
+  $backups = Get-Backups -Path $Path | Sort-Object -Property CreationTime
+  if ($backups.Length -ge 1) {
+    $lastBackupTime = $backups[0].CreationTime
+    $now = Get-Date
+    if ($now - $lastBackupTime -gt $Config.MaxBackupInterval) {
+      rotate
+      return
+    }
+  }
+}
+
+# Rotates the log files at directory $Path that match pattern.
+function Rotate-Files {
+  param (
+    # Pattern that file names must match to be rotated. Does not include parent path.
+    [parameter(Mandatory=$true)] [string]$Pattern,
+    # Path to the log directory containing files to rotate.
+    [parameter(Mandatory=$true)] [string]$Path,
+    # Config for file rotation.
+    [parameter(Mandatory=$true)] [FileRotationConfig]$Config
+
+  )
+  $files = Get-ChildItem -File -Path $Path | Where-Object { $_.Name -match $Pattern }
+  ForEach ($file in $files) {
+    Rotate-File -Path $file.FullName -Config $Config
+  }
+}
+
+# Schedules periodic log rotation with the Windows Task Scheduler
+function Schedule-LogRotation {
+  param (
+    # Pattern that file names must match to be rotated. Does not include parent path.
+    [parameter(Mandatory=$true)] [string]$Pattern,
+    # Path to the log directory containing files to rotate.
+    [parameter(Mandatory=$true)] [string]$Path,
+    # Interval at which to check logs against rotation criteria.
+    # Minimum 1 minute, maximum 31 days (see https://docs.microsoft.com/en-us/windows/desktop/taskschd/taskschedulerschema-interval-repetitiontype-element).
+    [parameter(Mandatory=$true)] [TimeSpan]$RepetitionInterval,
+    # Config for file rotation.
+    [parameter(Mandatory=$true)] [FileRotationConfig]$Config
+  )
+  # Write a powershell script to a file that imports this module ($PSCommandPath)
+  # and calls Rotate-Files with the configured arguments.
+  $scriptPath = "C:\rotate-kube-logs.ps1"
+  New-Item -Force -ItemType file -Path $scriptPath | Out-Null
+  Set-Content -Path $scriptPath @"
+Import-Module -Force ${PSCommandPath}
+`$maxBackupInterval = New-Timespan -Days $($Config.MaxBackupInterval.Days) -Hours $($Config.MaxBackupInterval.Hours) -Minutes $($Config.MaxBackupInterval.Minutes) -Seconds $($Config.MaxBackupInterval.Seconds)
+`$config = New-FileRotationConfig -Force:`$$($Config.Force) -MaxBackupInterval `$maxBackupInterval -MaxSize $($Config.MaxSize) -MaxBackups $($Config.MaxBackups)
+Rotate-Files -Pattern `"${Pattern}`" -Path `"${Path}`" -Config `$config
+"@
+  # The task will execute the logrotate.ps1 script created above.
+  # We explicitly set -WorkingDirectory to $Path for safety's sake, otherwise
+  # it runs in %windir%\system32 by default, which sounds dangerous.
+  $action = New-ScheduledTaskAction -Execute "powershell" -Argument "-NoLogo -NonInteractive -File ${scriptPath}" -WorkingDirectory $Path
+  # Start the task immediately, and trigger the task once every hour.
```

Comment:
Is this comment ("once every hour") still right?

Reply:
It should move to the other file, thanks